app/controllers/api/v1/tasks_controller.rb

```diff
@@ -38,48 +38,35 @@ def create
   api :PUT, '/tasks/:id', I18n.t('docs.tasks.actions.update')
   param :id, :number, I18n.t('docs.tasks.params.id'), required: true
   param :position, :number, I18n.t('docs.tasks.params.position')
+  param :done, :boolean, I18n.t('docs.tasks.params.done')
   param_group :task
   def update
     result = UpdateTask.call(task: task, params: task_params)
-    return render json: result.task, status: :created if result.success?
+    return render json: result.data, status: :ok if result.success?
```

Comment:
There I return an empty data or a message that all tasks complete if there was a `:done` parameter in `params`

Reply:
Is it required in spec to return the message that all tasks are done?

Reply:
According to tech task it is required.
[Link to UI](https://projects.invisionapp.com/share/UMB7L5XPF#/screens/227965767)
![Screenshot from 2019-12-22 13-03-10](https://user-images.githubusercontent.com/25724267/71320946-a2616e80-24bb-11ea-8861-4511d4af4fe8.png)

Reply:
Thank you for clarification.
You don't need to calculate it in the back end. As I mentioned above you have to return only new data to the front-end.

For example
front end received a list of tasks
```
tasks :
 [
  { id: 15, done: false },
  { id: 19, done: true },
  { id: 24, done: true },
  { id: 37, done: true },
 ]
```
When user would like to mark task 15 as completed the front end send you PUT request with data `id: 15, done: true` 
You update this task and if everything is alrigth you return `200 OK`
Frontend received this message, update task 15 and checks if all tasks are completed.

So, this calculation should be implemented on the front-end. 
We faced this misunderstanding because of this spec for a single page application "todo list" which includes front-end and serverside. In your case, you have to implement only serverside